planner/core/task.go

```diff
@@ -682,6 +683,53 @@ func (p *basePhysicalAgg) newPartialAggregate() (partial, final PhysicalPlan) {
 		groupByItems = append(groupByItems, gbyCol)
 	}
 
+	// Optimize
+	// Remove unnecessary schema column.
+	// When the select column is same with the group by key, the column can be removed and gets value from the group by key.
+	// e.g
+	// select a, count(b) from t group by a;
+	// The schema is [firstrow(a), count(b), a]. The column firstrow(a) is unnecessary.
+	// Can optimize the schema to [count(b), a] , and change the index to get value.
+	partialCursor = 0
+	partialAggNum := 0
+	partialAggFuncs := make([]*aggregation.AggFuncDesc, len(p.AggFuncs))
```

Comment:
1. `partialAggFuncs := make([]*aggregation.AggFuncDesc, 0, len(p.AggFuncs))`
2. `partialAggNum` can be removed
3. we can use `partialAggFuncs = append(partialAggFuncs, aggFunc)` in line 720.
4. line 726 can be removed

Reply:
update.